Sources_v3/StreamChatUI/Common Views/AvatarView.swift

```diff
@@ -62,7 +48,7 @@ open class AvatarView: UIView {
 
 // MARK: - AppearanceSetting
 
-extension AvatarView: AppearanceSetting {
+extension AvatarView {
     public class func initialAppearanceSetup(_ view: AvatarView) {
         view.defaultIntrinsicContentSize = .init(width: 40, height: 40)
     }
```

Comment:
Unfortunately, that won't work. The `OverridableView.initialAppearanceSetup` will be invoked in all the cases instead the method in the subclass.

To have `CustomView.initialAppearanceSetup` working it should be `override class` + some magic for getting the subclass type in run-time (via **type(of: self)**) + optional cast `OverridableView -> CustomView` in all the subclasses (since the method has to have the same signature to be overridden)

Reply:
Good point, thanks for the heads up! Do you think handling of `AppearanceSetting` should be in the base class at all?

The solution requires:
```swift
open class AvatarView: OverridableView {
    override public class func initialAppearanceSetup(_ view: OverridableView) {
        guard let view = view as? AvatarView else {
            return
        }
        view.defaultIntrinsicContentSize = .init(width: 40, height: 40)
    }
...
```
for every subclass we'll have, which is very repetitive.

Reply:
Addendum: casting is only required to access custom properties. If a subclass only styles default properties (from `UIView`) it'll work

Reply:
I'd move `AppearanceSetting` to subclasses. That requires calling `applyDefaultAppearance` in overridden `setupAppearance` but no optional casting in there + if end-users will subclass our custom view, better to have our custom view type in `class func initialAppearanceSetup` other than `OverridableView/OverridableControl`